### PR TITLE
txnstorage, txnengine: optimizations and fixes

### PR DESCRIPTION
**Pull request by reusee**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:
txnstorage: add Row.Value
txnstorage: remove Table.Index, rename IndexEntries to Index
txnstorage: merge PhysicalRow, MVCC, MVCCValues to PhysicalRow, Version
txnstorage: remove Row type param from Iter and TableIter
txnstorage: add Version.ID; fix Table.Iter
txnstorage: use btree path hinting
txnstorage, txnengine: add ID type
txnstorage: use slice instead of map in DataValue
txnstorage: add table benchmark
txnstorage: use int64 ID
txnstorage: optimize locking of Table
txnstorage: fix log tail
txnengine: add Request and Response enum
txnengine: remove ErrExisted
txnengine: remove ErrDatabaseNotFound
txnengine: remove ErrRelationNotFound
txnengine: remove ErrDefNotFound
txnengine: remove ErrIterNotFound
txnengine: remove ErrColumnNotFound, ErrReadOnly
txnengine: remove error reflection in DoTxnRequest

### CODE REVIEW

**Review by yingfeng — APPROVED** [review]

**Comment by fengttt** [comment]
Does this need someone from TAE team to take a look?

**Comment by reusee** [comment]
> Does this need someone from TAE team to take a look?

If the TAE team has time to do reviews, yes.
It seems Mochen wants to use Table in CN. He may be more concerned about the performance of this.

**Comment by nnsgmsone** [comment]
@fengttt I look forward to the early merging of this pr, after all, there are still some differences between tae and memengine, and the early elimination of these differences should be helpful to the speed of work.

**Review by zhangxu19830126 — APPROVED** [review]

**Comment by sukki37** [comment]
@mergifyio unqueue

**Comment by reusee** [comment]
@Mergifyio enqueue

**Comment by reusee** [comment]
@Mergify refresh

File: pkg/txn/storage/txn/catalog.go [hunk 1]
```diff
@@ -61,7 +71,7 @@ func (d DatabaseRow) AttrByName(tx *Transaction, name string) (ret Nullable, err
 			if attr.Name != name {
 				continue
 			}
-			if !typeMatch(ret.Value, attr.Type.Oid) {
+			if !memtable.TypeMatch(ret.Value, attr.Type.Oid) {
 				panic(fmt.Errorf("%s should be %v typed", name, attr.Type))
```
Comment (fengttt) [inline pkg/txn/storage/txn/catalog.go:75]: Clean up old code when you see one. :)

Comment (reusee) [reply pkg/txn/storage/txn/catalog.go]: These are for panic, I will use Sprintf instead.

**Pull request merged**